### ISSUE TITLE
Fail fast if user provides unknown flags

### ISSUE THREAD

**Issue by rauchg** [vercel/vercel#8]
To avoid dangerous and costly mistakes.

**Comment by codyzu** [comment]
How do you guys feel about switching from minimist to [commander](https://www.npmjs.com/package/commander) for the options parsing?

I think this will help with #3, #8, and  #9. I think we can have better "fail fast" functionality and more descriptive code by moving to commander. It should remove a lot of the manually verification of options.

I'm working on an idea for #3 with commander, but I wanted to see if there were any objections before I put together a pull request...

**Comment by OlliV** [comment]
I have been playing around with commander and it feels pretty good and simple to use. @leo @rauchg what do you think?

**Comment by leo** [comment]
Commander might be the industry's most used package for this purpose, but it's actually full of old code and bloated with unnecessary stuff. Let's go with [args](https://github.com/leo/args) (like now-serve and list)! 😊

**Comment by leo** [comment]
Is anyone interested in switching the project to args? Because that would it make it a lot easier for me to merge now-serve into now. If not, I'll tacke it!

**Comment by codyzu** [comment]
@leo I've been looking at this again. I'm looking at how it could work with [args](https://github.com/leo/args), but I don't see an obvious way to catch unknown flags. I see minimist has the `unknown` config param that is called for unknown params, but it's not (yet) propagated into args.

Am I missing something? Or is this something we would want to add to args?

**Comment by leo** [comment]
@codyzu Looks like a great addition to me! Also: https://github.com/leo/args/issues/45

**Comment by rauchg** [comment]
We started using commander, but for example we didn't have the ability to override quitting to make it wait on the update checker. It's very opinionated and tends to be quite inflexible sometimes.

**Comment by rauchg** [comment]
tl;DR: I'd rather re-implement checking for unknown flags rather than trade off flexibility

**Comment by codyzu** [comment]
I assumed there must be a back story behind not using commander. That makes sense.

Thanks @leo, looks like leo/args#45 is the way to go. I'll look at that first...

**Comment by rauchg** [comment]
Replaced by #1171
